### PR TITLE
[SYNCOPE-1657] Changes to Topology for allowing to specify a resource name during clone

### PR DESCRIPTION
**Pull request by eeleren**

### CODE REVIEW

**Comment by ilgrosso** [comment]
@eeleren thanks for your contribution
If you haven't done that yet, please fill and send your [ICLA](http://www.apache.org/licenses/#clas)

File: client/idm/console/src/main/resources/org/apache/syncope/client/console/topology/TopologyTogglePanel.properties [hunk 1]
```diff
@@ -23,6 +23,7 @@ connector.menu.history=Configuration history
 
 resource.new=New resource
 resource.edit=Edit resource {0}
+resource.clone=Clone resource {0}
```
Comment (ilgrosso) [inline client/idm/console/src/main/resources/org/apache/syncope/client/console/topology/TopologyTogglePanel.properties:26]: This must be added to all languages, e.g. `TopologyTogglePanel_*.properties`

File: client/idm/console/src/main/java/org/apache/syncope/client/console/topology/TopologyTogglePanel.java [hunk 2]
```diff
@@ -627,44 +626,41 @@ public String getAjaxIndicatorMarkupId() {
...
+                        modelObject.getOrgUnit().setKey(null);
+                        for (ItemTO item : modelObject.getOrgUnit().getItems()) {
                             item.setKey(null);
                         }
                     }
-                    for (ProvisionTO provision : resource.getProvisions()) {
+                    for (ProvisionTO provision : modelObject.getProvisions()) {
                         provision.setKey(null);
                         if (provision.getMapping() != null) {
+
```
Comment (ilgrosso) [inline client/idm/console/src/main/java/org/apache/syncope/client/console/topology/TopologyTogglePanel.java:650]: Please remove this formatting-only change

File: client/idm/console/src/main/java/org/apache/syncope/client/console/topology/TopologyTogglePanel.java [hunk 3]
```diff
@@ -627,44 +626,41 @@ public String getAjaxIndicatorMarkupId() {
         fragment.add(history);
 
         // [SYNCOPE-1161] - Option to clone a resource
-        AjaxLink<String> clone = new IndicatingOnConfirmAjaxLink<>("clone", "confirmClone", true) {
+        // [SYNCOPE-1657] - Unable to define a new name for a cloned resource
```
Comment (ilgrosso) [inline client/idm/console/src/main/java/org/apache/syncope/client/console/topology/TopologyTogglePanel.java:629]: No need to place a reference to SYNCOPE-1657, please remove this line

File: client/idm/console/src/main/java/org/apache/syncope/client/console/topology/TopologyTogglePanel.java [hunk 4]
```diff
@@ -627,44 +626,41 @@ public String getAjaxIndicatorMarkupId() {
...
+        AjaxLink<String> clone = new IndicatingAjaxLink<>("clone") {
 
             private static final long serialVersionUID = -7978723352517770644L;
 
             @Override
             public void onClick(final AjaxRequestTarget target) {
                 try {
-                    ResourceTO resource = ResourceRestClient.read(node.getKey());
-                    resource.setKey("Copy of " + resource.getKey());
+                    ResourceTO modelObject = ResourceRestClient.read(node.getKey());
```
Comment (ilgrosso) [inline client/idm/console/src/main/java/org/apache/syncope/client/console/topology/TopologyTogglePanel.java:637]: Please name this object back to `resource`, which improves readability

**Review by ilgrosso — CHANGES_REQUESTED** [review]

**Comment by eeleren** [comment]
> @eeleren thanks for your contribution If you haven't done that yet, please fill and send your [ICLA](http://www.apache.org/licenses/#clas)

ICLA sent

**Pull request merged**